report/src/main/webapp/app/services/metadata.service.js

```diff
@@ -89,10 +91,38 @@ define(['angularAMD', 'metadataCacheService', 'metadataEndpointService'],
           return step;
         }
 
+        function getStepByStepName(testName, urlName, stepName) {
+          var step = null;
+          var url = getUrl(testName, urlName);
+          for (var stepNr = 0; stepNr < url.steps.length; stepNr++) {
+            if (url.steps[stepNr].name === stepName) {
+              step = url.steps[stepNr];
+            }
+          }
+          return step;
+        }
+
+        function getStepComparator(testName, urlName, stepName, comparatorIndex) {
+          var step = getStepByStepName(testName, urlName, stepName);
+          var comparator = null;
+          if (step !== null && step.comparators[comparatorIndex] !== null ) {
+            comparator = step.comparators[comparatorIndex];
+          }
+          return comparator
+        }
+
         function notifyMetadataChanged() {
           $rootScope.$emit('metadata:changed');
         }
 
+        function updateComment(vm, param) {
+          var caseName = vm.model.displayName.split(" ")[0];
```

Comment:
Is that logic applied anywhere else? If yes, maybe it would be worth to put it into one place and reuse.